### DISCUSSION TITLE
Proposal to export custom properties of .astro components through Astro.glob()

### DISCUSSION THREAD

**Discussion by ramonchu** [withastro/roadmap#368] [Proposal]
Unless I'm wrong due to ignorance, I can't find a way to export custom properties in .astro components that can be consumed via Astro.glob(...). For example:

// Component.astro
---
export customProperty = "customValue";
---
<div>html of component... </div>

// index.astro ----
---
const services = await Astro.glob('./services/*.astro');
// assume services.length >0

services.map(service => console.log(service.customProperty));

---

**Comment by reggi** [comment]
I kind of want to export in front-matter as well but is tricky because you can `return`

**Comment by florian-lefebvre** [comment]
Closing because this is outdated. Astro.glob is deprecated
